htmresearch/frameworks/nlp/imbu.py

```diff
@@ -147,27 +147,32 @@ def _modelFactory(self, modelName, savePath, **kwargs):
     """ Imbu model factory.  Returns a concrete instance of a classification
     model given a model type name and kwargs.
     """
-    modelType = (
-      getattr(ClassificationModelTypes, modelName) or self.defaultModelType )
+    if modelName not in {
```

Comment:
Since this set is the same as is considered below, you could add a final `else` clause that simply raises the `ValueError` exception.